src/app/core/entity-components/entity-list/entity-list.component.spec.ts

```diff
@@ -256,4 +257,39 @@ describe("EntityListComponent", () => {
     expect(component.filtersConfig).toEqual([]);
     expect(component.filterSelections).toEqual([]);
   });
+
+  it("should sort standard objects", () => {
+    const children = [
+      new Child("0"),
+      new Child("1"),
+      new Child("2"),
+      new Child("3"),
+    ];
+    children[0].name = "AA";
+    children[3].name = "AB";
+    children[2].name = "Z";
+    children[1].name = "C";
+    component.entityList = children;
+    component.sort.sort({ id: "name", start: "asc", disableClear: false });
+    const sortedIds = component.entityDataSource
+      .sortData(children, component.sort)
+      .map((value) => value.getId());
+    expect(sortedIds).toEqual(["0", "3", "1", "2"]);
+  });
+
+  it("should sort non-standard objects", () => {
+    const notes = [new Note("0"), new Note("1"), new Note("2"), new Note("3")];
+    notes[0].category = { id: "0", label: "AA" };
+    notes[3].category = { id: "1", label: "AB" };
+    notes[2].category = { id: "2", label: "Z" };
+    notes[1].category = { id: "3", label: "C" };
+    component.ngOnInit();
+    component.entityList = notes;
+    component.sort.sort({ id: "category", start: "asc", disableClear: false });
+    const sortedIds = component.entityDataSource
+      .sortData(notes, component.sort)
+      .map((note) => note.getId());
+    console.log(sortedIds);
+    expect(sortedIds).toEqual(["0", "3", "1", "2"]);
```

Comment:
maybe we could make this test code reusable. I was looking into "parametrized tests" recently: https://stackoverflow.com/questions/5061512/whats-a-good-way-to-reuse-test-code-using-jasmine

Reply:
In general, I think we could profit from some test reusability. For example, it often occurs that one needs an array of entities with certain attributes set (e.g. an array of Notes that all have a certain title). This yields the same 5-10 lines of code in a lot of places. Maybe we should consider refactoring some of the tests to make them look more convenient as well as make further test more easy?

Reply:
yeah, absolutely agree.  Will merge this now to clear the path for the big Angular upgrade but we should look into test code improvements